### DISCUSSION TITLE
Pyenv not working; Build keeps failing; pyenv doctor command says build keeps failing.

### DISCUSSION THREAD

**Discussion by ghost** [pyenv/pyenv#2745] [Q&A] [answered]
Hello, I am trying to set up a virtual environment using homebrew on mac ventura through zsh terminal. I have tried reinstalling homebrew, xcode, pyenv and the sort. I have also looked for any similar discussion pages online and have tried commands that have worked for other people, but unfortunately hasnt worked for me. I am also relatively new in coding and navigating the computer through terminal, so any tips or solutions to this would be very helpful.

I also have looked into my $PATH, but im not sure if it is related to my issue, and I have looked online and couldnt find resources to help with the $PATH issue. 

Here is the error that I get when I try installing python 3.8.15 through pyenv: 

<img width="1723" alt="Screenshot 2023-07-23 at 3 11 31 PM" src="https://github.com/pyenv/pyenv/assets/131822191/5e8b6b15-ec1a-41c1-abe4-b98e93243af3">

Pyenv doctor command is also not working, here is the error code:

<img width="1260" alt="Screenshot 2023-07-23 at 3 12 29 PM" src="https://github.com/pyenv/pyenv/assets/131822191/b187fff3-2369-4609-bd2a-609fbb2fe99a">

Here is what I get when I put this command into the terminal: 

echo $PATH | grep --color=auto "$(pyenv root)/shims"

<img width="1719" alt="Screenshot 2023-07-23 at 3 15 10 PM" src="https://github.com/pyenv/pyenv/assets/131822191/52c240be-d3e0-4ab4-89ec-64d60ee53ca9">

Does this mean that .pyenv is in my $PATH and in the correct spot? I would appreciate if someone explained this to me so I can understand what to look for in the future if a problem similar to this arrises. 

I have been genuinely stuck for the past couple of days and have reached out to some of my comp sci friends, but they havent been able to help me with this. I decided to post on this discussion page in hopes of getting this issue resolved.

**Comment by native-api** [comment, marked as answer]
As per https://github.com/pyenv/pyenv/issues/2707, this suggests incompatibilities of between installed Homebrew modules and the OS -- which can happen after an OS upgrade and/or migration to Apple Silicon.

A clean reinstall of [Homebrew](https://docs.brew.sh/FAQ#how-do-i-uninstall-homebrew) and [Pyenv](https://github.com/pyenv/pyenv#uninstalling-pyenv) is indeed suggested.

Note that if you're on Apple Silicon, there can be 2 Homebrew installations -- under `/usr/local` (x64) and under `/opt/homebrew` (arm64), check both locations.
To keep things simple, you should normally work in Arm64 architecture ("`arch`" should return "`arm`", "`arm64`" or "`aarch64`").
